<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# typescript/no-useless-empty-export

<div class="rule-meta">
<span class="default-on">
‚úÖ This rule is turned on by default.
</span>
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule.
</span>
</div>

### What it does

Disallow empty exports that don't change anything in a module file.

### Example

### Bad

```ts
export const value = "Hello, world!";
export {};
```

### Good

```ts
export const value = "Hello, world!";
```
