<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# vitest/prefer-to-be-truthy

<div class="rule-meta">
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule.
</span>
</div>

### What it does

This rule warns when `toBe(true)` is used with `expect` or `expectTypeOf`. With `--fix`, it will be replaced with `toBeTruthy()`.

### Examples

```javascript
// bad
expect(foo).toBe(true);
expectTypeOf(foo).toBe(true);

// good
expect(foo).toBeTruthy();
expectTypeOf(foo).toBeTruthy();
```
