<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-length-as-slice-end

<div class="rule-meta">
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule.
</span>
</div>

### What it does

Disallow using `length` as the end argument of a `slice` call.

### Why is this bad?

Passing `length` as the end argument of a `slice` call is unnecessary and can be confusing.

### Example

```javascript
// Bad
foo.slice(1, foo.length);

// Good
foo.slice(1);
```
