<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-new-array

<div class="rule-meta">
<span class="default-on">
âœ… This rule is turned on by default.
</span>
</div>

### What it does

Disallow `new Array()`.

### Why is this bad?

When using the `Array` constructor with one argument, it's not clear whether the argument is meant to be the length of the array or the only element.

### Example
```javascript
// bad
const array = new Array(1);
const array = new Array(42);
const array = new Array(foo);

// good
const array = Array.from({ length: 42 });
const array = [42];
```

