<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/no-single-promise-in-promise-methods

<div class="rule-meta">
<span class="default-on">
‚úÖ This rule is turned on by default.
</span>
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule for some violations.
</span>
</div>

### What it does

Disallow passing single-element arrays to Promise methods

### Why is this bad?

Passing a single-element array to `Promise.all()`, `Promise.any()`, or `Promise.race()` is likely a mistake.

### Example

Bad

```js
const foo = await Promise.all([promise]);
const foo = await Promise.any([promise]);
const foo = await Promise.race([promise]);
const promise = Promise.all([nonPromise]);
```

Good

```js
const foo = await promise;
const promise = Promise.resolve(nonPromise);
const foo = await Promise.all(promises);
const foo = await Promise.any([promise, anotherPromise]);
const [{ value: foo, reason: error }] = await Promise.allSettled([promise]);
```
