<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-set-size

<div class="rule-meta">
<span class="default-on">
âœ… This rule is turned on by default.
</span>
<span class="fix">
ðŸš§ An auto-fix is still under development.
</span>
</div>

### What it does

Prefer `Set#size` over `Set#length` when the `Set` is converted to an array.

### Why is this bad?

Using `Set#size` is more readable and performant.

### Example

```javascript
// bad
const length = [...new Set([1, 2, 3])].length;

// good
const size = new Set([1, 2, 3]).size;
```
