<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-modern-math-apis

<div class="rule-meta">
</div>

### What it does

Checks for usage of legacy patterns for mathematical operations.

### Why is this bad?

Modern JavaScript provides more concise and readable alternatives to legacy patterns.

Currently, the following cases are checked:

- Prefer `Math.log10(x)` over alternatives
- Prefer `Math.hypot(â€¦)` over alternatives

### Example

```javascript
// Bad
Math.log(x) * Math.LOG10E;
Math.sqrt(a * a + b * b);

// Good
Math.log10(x);
Math.hypot(a, b);
```
