<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-node-protocol

<div class="rule-meta">
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule.
</span>
</div>

### What it does

Prefer using the `node:protocol` when importing Node.js built-in modules

### Example

```javascript
// Bad
import fs from "fs";
// Good
import fs from "node:fs";
```
