<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-string-starts-ends-with

<div class="rule-meta">
<span class="default-on">
‚úÖ This rule is turned on by default.
</span>
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule.
</span>
</div>

### What it does

Prefer [`String#startsWith()`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith) and [`String#endsWith()`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith) over using a regex with `/^foo/` or `/foo$/`.

### Why is this bad?

Using `String#startsWith()` and `String#endsWith()` is more readable and performant as it does not need to parse a regex.

### Example
```javascript
// Bad
const foo = "hello";
/^abc/.test(foo);

// Good
const foo = "hello";
foo.startsWith("abc");
```

