<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/prefer-optional-catch-binding

<div class="rule-meta">
<span class="fix">
ðŸš§ An auto-fix is still under development.
</span>
</div>

### What it does

Prefers omitting the catch binding parameter if it is unused

### Why is this bad?

It is unnecessary to bind the error to a variable if it is not used.

### Example

```javascript
// Bad
try {
  // ...
} catch (e) {}

// Good
try {
  // ...
} catch {}
```
