<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# unicorn/require-array-join-separator

<div class="rule-meta">
</div>

### What it does

Enforce using the separator argument with Array#join()

### Why is this bad?

It's better to make it clear what the separator is when calling Array#join(),
instead of relying on the default comma (',') separator.

### Example

```javascript
// Bad
foo.join();

// Good
foo.join(",");
```
