<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# oxc/bad-replace-all-arg

<div class="rule-meta">
<span class="default-on">
âœ… This rule is turned on by default.
</span>
</div>

### What it does

This rule warns when the `replaceAll` method is called with a regular expression that does not have the global flag (g).

### Why is this bad?

The `replaceAll` method replaces all occurrences of a string with another string. If the global flag (g) is not used in the regular expression, only the first occurrence of the string will be replaced.

### Example
```javascript
// Bad: The global flag (g) is missing in the regular expression.
withSpaces.replaceAll(/\s+/, ',');

// Good: The global flag (g) is used in the regular expression.
withSpaces.replaceAll(/\s+/g, ',');
```

