<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# oxc/double-comparisons

<div class="rule-meta">
<span class="default-on">
âœ… This rule is turned on by default.
</span>
</div>

### What it does

This rule checks for double comparisons in logical expressions.

### Why is this bad?

Redundant comparisons can be confusing and make code harder to understand.

### Example

```javascript
// Bad
x === y || x < y;
x < y || x === y;

// Good
x <= y;
x >= y;
```
