<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/no-sparse-arrays <Badge type="info" text="Correctness" />

<div class="rule-meta">
<Alert class="default-on" type="success">
<span class="emoji">✅</span> This rule is turned on by default.
</Alert>
</div>

### What it does

Disallow sparse arrays.

### Why is this bad?

The confusion around sparse arrays is enough that it’s recommended to avoid using them unless you are certain that they are useful in your code.

### Example

```javascript
var items = [, ,];
var colors = ["red", , "blue"];
```

## How to use

To **enable** this rule in the CLI or using the config file, you can use:

::: code-group

```bash [CLI]
oxlint --deny /no-sparse-arrays
```

```json [Config (.oxlintrc.json)]
{
  "rules": {
    "/no-sparse-arrays": "error"
  }
}
```

:::

## References

- [Rule Source](https://github.com/oxc-project/oxc/blob/85b14a378b63d1839da9f3a11e14db5a7fddb472/crates/oxc_linter/src/rules/eslint/no_sparse_arrays.rs)
