<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# eslint/valid-typeof

<div class="rule-meta">
<span class="default-on">
‚úÖ This rule is turned on by default.
</span>
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule for some violations.
</span>
</div>

### What it does

Enforce comparing `typeof` expressions against valid strings

### Why is this bad?

It is usually a typing mistake to compare the result of a `typeof`
operator to other string literals.

### Example

```js
// requireStringLiterals: false
// incorrect:
typeof foo === "strnig";
// correct:
typeof foo === "string";
typeof foo === baz;

// requireStringLiterals: true
// incorrect:
typeof foo === baz;
```
