<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# promise/no-new-statics

<div class="rule-meta">
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule.
</span>
</div>

### What it does

Disallow calling new on a Promise static method.

### Why is this bad?

Calling a Promise static method with new is invalid, resulting in a TypeError at runtime.

### Example
```javascript
new Promise.resolve(value);
```

