<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/aria-props

<div class="rule-meta">
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule for some violations.
</span>
</div>

### What it does

Enforces that elements do not use invalid ARIA attributes.

### Why is this bad?

Using invalid ARIA attributes can mislead screen readers and other assistive technologies.
It may cause the accessibility features of the website to fail, making it difficult
for users with disabilities to use the site effectively.

This rule includes fixes for some common typos.

### Example

```javascript
// Bad
<input aria-labeledby="address_label" />

// Good
<input aria-labelledby="address_label" />
```
