<!-- This file is auto-generated by tasks/website/src/linter/rules/doc_page.rs. Do not edit it manually. -->

# jsx_a11y/no-autofocus

<div class="rule-meta">
<span class="fix">
üõ†Ô∏è An auto-fix is available for this rule.
</span>
</div>

### What it does
Enforce that autoFocus prop is not used on elements. Autofocusing elements can cause usability issues for sighted and non-sighted users, alike.

### Rule Option
This rule takes one optional object argument of type object:

```
{
    "rules": {
        "jsx-a11y/no-autofocus": [ 2, {
            "ignoreNonDOM": true
        }],
    }
}
```

For the `ignoreNonDOM` option, this determines if developer created components are checked.

### Example
// good

```javascript
<div />
```

// bad

```
<div autoFocus />
<div autoFocus="true" />
<div autoFocus="false" />
<div autoFocus={undefined} />
```


